api/v1alpha4/azuremachine_validation.go

```diff
@@ -221,11 +211,10 @@ func ValidateDataDisksUpdate(oldDataDisks, newDataDisks []DataDisk, fieldPath *f
 	return allErrs
 }
 
-func validateDiffDiskSetings(d *DiffDiskSettings, fldPath *field.Path) *field.Error {
+func validateDiffDiskSettings(d *DiffDiskSettings, fldPath *field.Path) *field.Error {
 	if d != nil {
```

Comment:
```suggestion
	if d != nil && d.Option != string(compute.Local) {
```

Reply:
Is this check even needed since we already have kubebuilder validation annotation on the field? https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/adaf9c6d4b03bbe42174ad2dd1a7c0fba66a56dd/api/v1alpha4/types.go#L405

Reply:
you're right, removed it